builtin-features/operations.inc

```diff
@@ -25,10 +25,39 @@ packToken Assign(const packToken& left, const packToken& right, evaluationData*
       // It is not possible to assign directly to
       // the global scope. It would be easy for the user
       // to do it by accident, thus:
-      if (!map || *map == TokenMap::default_global()) {
+      if (!map || *map == TokenMap::default_global()) 
+      {
         data->scope[var_name] = right;
-      } else {
-        (*map)[var_name] = right;
+      }
+      else 
+      {
+          packToken* assignee = map->find(var_name);
```

Comment:
Since map was a non-null output from a call to: `.findMap(var_name)` it is safe to assume that this key is present on this map. So it might be better to use:

```suggestion
          packToken assignee = (*map)[var_name]; // This would throw if no value was found
```

Then you could remove the `if (assignee)` below

Reply:
right, I was leaning on the safe side when I wrote that.